File: Day8/Solution.swift
// Enter your code here. Read input from STDIN. Print output to STDOUT

var phoneBook: [String:String] = [:]
let n = Int(readLine()!)!

for i in 1...n {
    let currentLine = readLine()!.split(separator: " ")
    phoneBook[String(currentLine[0])] = String(currentLine[1])
}

while(1==1) {
    if let query = readLine() {
        if(phoneBook[query] != nil) {
            print(query + "=" + phoneBook[query]!)
        } else {
            print("Not found")
        }
    } else {
        break
    }
}

File: Day8/Solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Scan()
	n, _ := strconv.Atoi(in.Text())

	phoneBook := make(map[string]int)

	for i := 0; i < n; i++ {
		in.Scan()
		input := in.Text()
		values := strings.Split(input, " ")
		name := values[0]
		phone, _ := strconv.Atoi(values[1])

		phoneBook[name] = phone
	}

	for {
		in.Scan()
		key := in.Text()
		if key == "" {
			break
		}

		if value, ok := phoneBook[key]; ok {
			fmt.Printf("%s=%d\n", key, value)
		} else {
			fmt.Println("Not found")
		}
	}
}
